src/core_crypto/guassian_sampler.rs

```diff
@@ -0,0 +1,100 @@
+use rand::SeedableRng;
+use rand_chacha::ChaCha20Rng;
+use rand_distr::{Distribution, Normal};
+
+/// `TruncatedDiscreteGaussian` represents the parameters of a truncated discrete Gaussian distribution with:
+/// * standard deviation `sigma`
+/// * bounds `[-bound, bound]`, where bound is equal to `(6*sigma).round()`
+/// * mean 0
+#[derive(Clone, Debug)]
+struct TruncatedDiscreteGaussian {
+    sigma: f64,
+    bound: i64,
+}
+
+#[derive(Clone, Debug)]
+struct GaussianSampler {
+    params: TruncatedDiscreteGaussian,
+    rng: ChaCha20Rng,
+}
+
+impl GaussianSampler {
+    /// `new` returns a new `GaussianSampler` with the given parameters starting from the standard deviation `sigma`.
+    pub fn new(sigma: f64) -> Self {
+        assert!(sigma > 0.0, "sigma must be positive");
+        let bound = (6.0 * sigma).round() as i64;
+        let rng = ChaCha20Rng::from_entropy();
+        Self {
+            params: TruncatedDiscreteGaussian { sigma, bound },
+            rng,
+        }
+    }
+
+    /// `sample` returns a sample from the truncated discrete Gaussian distribution.
+    /// The technique used is patterned after [lattigo](https://github.com/tuneinsight/lattigo/blob/c031b14be1fb3697945709d7afbed264fa845442/ring/sampler_gaussian.go#L71).
+    /// In particular, `sampled_val` is sampled from a normal distribution with mean 0 and standard deviation `sigma`.
+    /// If `sampled_val` is within the bounds, it is rounded and returned.
+    pub fn sample(&mut self) -> i64 {
+        let discrete_normal = Normal::new(0.0, self.params.sigma).unwrap();
+        let sampled_val = discrete_normal.sample(&mut self.rng);
+        if sampled_val.abs() < self.params.bound as f64 {
```

Comment:
Remove the check and the recursive call because it guarantees that sample is within $6\sigma$ with 100% probability, which leaks information and is insecure.

Reply:
I don't think there will be any problem. Value will be $> 6\sigma$ with very small probability and will not have much effect on noise growth. 

Moreover, we aren't shooting for tight bounds at present so bounding seems unnecessary.